src/pkg/build/assemblers/assembler_sif.go

```diff
@@ -129,3 +136,40 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) (err error) {
 
 	return
 }
+
+// changeOwner check the command being called with sudo with the environment
+// variable SUDO_COMMAND. Pattern match that for the singularity bin
+func changeOwner() (int, int, bool) {
+	r := regexp.MustCompile("(singularity)")
+	sudoCmd := os.Getenv("SUDO_COMMAND")
+	if !r.MatchString(sudoCmd) {
+		return 0, 0, false
+	}
+
+	if syscall.Getuid() == 0 && os.Getenv("SUDO_USER") == "root" {
```

Comment:
This logic here is incorrect.

First of all, we want to continue only if the UID == 0. Second, if `SUDO_USER` is set to *anything* we should continue.

I think the logic should look like this:
`if os.Getenv("SUDO_USER") == "" || syscall.Getuid() != 0 {...}`

This means we *only* proceed with the function *if and only if* `SUDO_USER` is *set* and `uid==0`.

Reply:
066de5d